packages/shared/testing/evm-based/src/lib/evm-based.erc721.test-factory.ts

```diff
@@ -427,6 +508,84 @@ export const erc721TestFactory = {
       })
     },
 
+    mintMultipleProvenanceSignedTransaction: (
+      sdk: SdkWithErc721Functions,
+      testData: BlockchainTestData,
+      chain: 'ETH' | 'MATIC' | 'KCS' | 'ONE' | 'BSC' | 'KLAY',
+    ) => {
+      const provider = testData?.PROVIDER
+      const address = testData.TESTNET.ERC_721?.ADDRESS
+        ? testData.TESTNET.ERC_721?.ADDRESS
+        : '0x811DfbFF13ADFBC3Cf653dCc373C03616D3471c9'
+
+      it('valid from privateKey', async () => {
+        const result = await sdk.prepare.mintMultipleProvenanceSignedTransaction(
+          {
+            to: [address, address],
+            contractAddress: testData.MAINNET.ERC_721!.CONTRACT_ADDRESS,
+            fromPrivateKey: testData.MAINNET.ERC_721!.PRIVATE_KEY,
+            tokenId: [new Date().getTime().toString(), new Date().getTime().toString()],
+            url: ['https://my_token_data.com', 'https://my_token_data.com'],
+            chain,
+            authorAddresses: [[address], [address]],
+            fixedValues: [['1'], ['2']],
+            fee: {
+              gasLimit: '3264521',
+              gasPrice: '20',
+            },
+          },
+          provider,
+        )
+        expectHexString(result)
+      })
+
+      it('valid from SignatureId', async () => {
+        const result = await sdk.prepare.mintMultipleProvenanceSignedTransaction(
+          {
+            to: [address, address],
+            contractAddress: testData.MAINNET.ERC_721!.CONTRACT_ADDRESS,
+            signatureId: 'cac88687-33ed-4ca1-b1fc-b02986a90975',
+            tokenId: [new Date().getTime().toString(), new Date().getTime().toString()],
+            url: ['https://my_token_data.com', 'https://my_token_data.com'],
+            chain,
+            fixedValues: [['1'], ['2']],
+            authorAddresses: [[address], [address]],
+            fee: {
+              gasLimit: '3264521',
+              gasPrice: '20',
+            },
+          },
+          provider,
+        )
+        const json = JSON.parse(result)
+        expectHexString(json.data)
+      })
+
+      it('invalid address', async () => {
+        try {
+          await sdk.prepare.mintMultipleProvenanceSignedTransaction(
+            {
+              to: ['invalid', 'invalid'],
+              contractAddress: testData.MAINNET.ERC_721!.CONTRACT_ADDRESS,
+              fromPrivateKey: testData.MAINNET.ERC_721!.PRIVATE_KEY,
+              tokenId: [new Date().getTime().toString(), new Date().getTime().toString()],
+              url: ['https://my_token_data.com', 'https://my_token_data.com'],
+              chain,
+              fixedValues: [['1'], ['2']],
+              fee: {
+                gasLimit: '3264521',
+                gasPrice: '20',
+              },
+            },
+            provider,
+          )
+          fail()
+        } catch (e: any) {
+          expect(e.reason).toMatch('invalid address')
```

Comment:
Take a look at 
packages/shared/testing/btc-based/src/lib/btc-based.tx.test-factory.ts

).rejects.toThrowSdkErrorWithCode(SdkErrorCode.BTC_FEE_TOO_SMALL)

Reply:
I tried implementing something similar but it was throwing me this error while testing
<img width="965" alt="image" src="https://user-images.githubusercontent.com/52315405/154633008-26ed01f8-2712-4915-a794-97783d524e3b.png">
The code in the error is INVALID_ARGUMENT but it's not working...